### PR TITLE
add an 'info' endpoint to the api akin to marathon's

### PR DESCRIPTION
**Pull request by Rob-Johnson**
up for early review and suggestions on what other info we might want here.

@tub @solarkennedy thoughts?

```
robj@robj-yelp-mb:~/workspace/chronos info-api % curl http://0.0.0.0:8081/info | jq .
{
  "schedulerConfiguration": {
    "defaultTaskEpsilon": 60,
    "hostname": "b905ec02e1ac",
    "schedulerHorizonSeconds": 60,
    "master": "zk://zookeeper:2181/mesos-testcluster",
    "failoverTimeoutSeconds": 604800,
    "declineOfferDuration": "None",
    "minReviveOffersInterval": 5000,
    "reconciliationInterval": 600,
    "leaderMaxIdleTime": 5000,
    "clusterName": "None",
    "user": "root"
  },
  "mail_config": {
    "mail_user": "None",
    "mail_from": "None",
    "mail_ssl_on": "None",
    "mail_server": "None",
    "mail_password": "None"
  },
  "frameworkId": "058f572c-9040-46f6-a86a-f8b37acabdeb-0000",
  "version": "2.5.0-yelp14-SNAPSHOT",
  "zookeeper_config": {
    "zookeeperTimeout": 10000,
    "zookeeperPath": "/chronos"
  }
}
```

### CODE REVIEW

File: src/main/scala/org/apache/mesos/chronos/scheduler/api/PathConstants.scala [hunk 1]
```diff
@@ -29,7 +29,6 @@ object PathConstants {
   final val taskBasePath = "/scheduler/task/"
   final val killTaskPattern = "kill/{jobName}"
 
-  final val isMasterPath = "isMaster"
```
Comment (Rob-Johnson) [inline src/main/scala/org/apache/mesos/chronos/scheduler/api/PathConstants.scala:32]: I did not mean to remove these.

File: src/main/scala/org/apache/mesos/chronos/scheduler/api/InfoResource.scala [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        "leaderMaxIdleTime" -> schedulerConfiguration.leaderMaxIdleTimeMs(),
+        "minReviveOffersInterval" -> schedulerConfiguration.minReviveOffersInterval(),
+        "declineOfferDuration" -> schedulerConfiguration.declineOfferDuration.get.getOrElse("None"),
+        "reconciliationInterval" -> schedulerConfiguration.reconciliationInterval(),
+        "defaultTaskEpsilon" -> schedulerConfiguration.taskEpsilon.get.getOrElse("defaultTaskEpsilon")
+    )
+    val mailConfig = Map(
+      "mail_server" -> schedulerConfiguration.mailServer.get.getOrElse("None"),
+      "mail_user" -> schedulerConfiguration.mailUser.get.getOrElse("None"),
+      "mail_password" -> schedulerConfiguration.mailUser.get.getOrElse("None"),
```
Comment (solarkennedy) [inline src/main/scala/org/apache/mesos/chronos/scheduler/api/InfoResource.scala]: I don't know if it is a super great idea to spit out the password

**Comment by solarkennedy** [comment]
Looks great!

**Comment by tub** [comment]
Framework name may be useful in there, can't really think of anything else.

**Comment by Rob-Johnson** [comment]
well, I added a bunch more. I still need to add the cassandra and graphite bits

```
{
  "mesos_config": {
    "mesos_checkpoint": true,
    "mesos_authentication_principal": "None",
    "master": "zk://zookeeper:2181/mesos-testcluster",
    "mesos_task_disk": 256,
    "mesos_role": "*",
    "mesos_task_mem": 128,
    "mesos_framework_name": "chronos-2.5.0-yelp14-SNAPSHOT",
    "mesos_task_cpu": 0.1
  },
  "mail_config": {
    "mail_server": "None",
    "mail_user": "None",
    "mail_from": "None",
    "mail_ssl_on": "None"
  },
  "frameworkId": "058f572c-9040-46f6-a86a-f8b37acabdeb-0000",
  "leader": "675aeaf6d12f:8081",
  "chronos_configuration": {
    "cluster_name": "None",
    "disable_after_failures": 0,
    "hostname": "675aeaf6d12f",
    "failure_retry_delay": 60000,
    "task_epsilon": 60,
    "reconciliation_interval": 600,
    "min_revive_offers_interval": 5000,
    "leader_max_idle_time": 5000,
    "default_task_epsilon": 60,
    "failover_timeout": 604800,
    "scheduler_horizon": 60,
    "decline_offer_duration": "None",
    "user": "root"
  },
  "version": "2.5.0-yelp14-SNAPSHOT",
  "zookeeper_config": {
    "zk_hosts": "zookeeper:2181",
    "zookeeper_timeout": 10000,
    "zookeeper_path": "/chronos"
  }
}
```

**Comment by Rob-Johnson** [comment]
eh, on second thought, I'll ditch the cassandra + graphite stuff. I'll come back to it if we ever use them.

**Comment by Rob-Johnson** [comment]
added some tests. @tub shipit?

File: src/main/scala/org/apache/mesos/chronos/scheduler/api/InfoResource.scala [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        "mesos_role" -> schedulerConfiguration.mesosRole(),
+        "mesos_framework_name" -> schedulerConfiguration.mesosFrameworkName(),
+        "mesos_authentication_principal" -> schedulerConfiguration.mesosAuthenticationPrincipal.get.getOrElse("None")
+    )
+
+    val mailConfig = Map(
+      "mail_server" -> schedulerConfiguration.mailServer.get.getOrElse("None"),
+      "mail_user" -> schedulerConfiguration.mailUser.get.getOrElse("None"),
+      "mail_from" -> schedulerConfiguration.mailFrom.get.getOrElse("None"),
+      "mail_ssl_on" -> schedulerConfiguration.mailFrom.get.getOrElse("None")
```
Comment (tub) [inline src/main/scala/org/apache/mesos/chronos/scheduler/api/InfoResource.scala]: mailFrom?

File: src/main/scala/org/apache/mesos/chronos/scheduler/api/PathConstants.scala (hunk 1 above)
Comment (tub) [inline src/main/scala/org/apache/mesos/chronos/scheduler/api/PathConstants.scala:32]: Looks like isMasterPath isn't used anyway... 👍

**Comment by tub** [comment]
Fix mail_ssl_on and :shipit:

**Pull request merged**